java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -99,25 +133,29 @@ public FieldVector getDataVector() {
 
   @Override
   public void setInitialCapacity(int numRecords) {
-    offsets.setInitialCapacity(numRecords + 1);
+    offsetAllocationSizeInBytes = (numRecords + 1) * OFFSET_WIDTH;
     vector.setInitialCapacity(numRecords * RepeatedValueVector.DEFAULT_REPEAT_PER_RECORD);
   }
 
   @Override
   public int getValueCapacity() {
-    final int offsetValueCapacity = Math.max(offsets.getValueCapacity() - 1, 0);
+    final int offsetValueCapacity = Math.max(getOffsetBufferValueCapacity() - 1, 0);
     if (vector == DEFAULT_DATA_VECTOR) {
       return offsetValueCapacity;
     }
     return Math.min(vector.getValueCapacity(), offsetValueCapacity);
   }
 
+  private int getOffsetBufferValueCapacity() {
+    return (int)((offsetBuffer.capacity() * 1.0)/OFFSET_WIDTH);
+  }
+
   @Override
   public int getBufferSize() {
-    if (getAccessor().getValueCount() == 0) {
+    if (getValueCount() == 0) {
       return 0;
     }
-    return offsets.getBufferSize() + vector.getBufferSize();
+    return ((valueCount + 1) * OFFSET_WIDTH) + vector.getBufferSize();
```

Comment:
Maybe refactor `(valueCount + 1) * OFFSET_WIDTH` to an function for readability?